src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,116 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY;
+    private static final float LOAD_FACTOR;
+    private static final int RESIZE_FACTOR;
+    private int currentCapacity;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    static {
+        INITIAL_CAPACITY = 16;
+        LOAD_FACTOR = 0.75f;
+        RESIZE_FACTOR = 2;
+    }
+
+    {
+        table = new Node[INITIAL_CAPACITY];
+        currentCapacity = INITIAL_CAPACITY;
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
+
+    private class Node<K, V> {
+        private final K key;
+        private V value;
+        private final int hashCode;
+        private Node<K, V> next;
+
+        private Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+            this.hashCode = setHashCode();
+            this.next = null;
+        }
+
+        private int setHashCode() {
+            return key != null ? Math.abs(key.hashCode() % currentCapacity) : 0;
+
+        }
+
+        private boolean ifEqualsKey(K key) {
+            return (this.key == key
+                    || this.key != null && this.key.equals(key));
+        }
+    }
+
+    private void resize() {
+        if (size >= threshold) {
+            currentCapacity *= RESIZE_FACTOR;
+            migrate();
+            threshold = (int) (currentCapacity * LOAD_FACTOR);
+        }
+    }
+
+    private void migrate() {
+        Node<K, V>[] tempTable = new Node[currentCapacity];
+        for (Node<K, V> currentNode : table) {
+            if (currentNode == null) {
+                continue;
+            }
+            Node<K, V> newNode = new Node<>(currentNode.key, currentNode.value);
+            tempTable[newNode.hashCode] = newNode;
+            if (currentNode.next != null) {
+                newNode.next = currentNode.next;
+            }
+        }
+        table = tempTable;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
```

Comment:
we mast resize table when we size >= threshold, but, as i understood, your methods does it every time